### PR TITLE
Check lkl_virtio_dev_setup() return value

### PR DESCRIPTION
**Pull request by Francis-Liu**
Part of fix to #370

### CODE REVIEW

File: src/lkl/setup.c [hunk 1]
```diff
@@ -861,7 +861,8 @@ void lkl_mount_disks(
     if (!root)
         sgxlkl_fail("No root disk provided. Aborting...\n");
 
-    lkl_add_disks(root, mounts, num_mounts);
+    if (lkl_add_disks(root, mounts, num_mounts) != 0)
```
Comment (davidchisnall) [inline src/lkl/setup.c:864]: `!= 0` is redundant here.  I normally prefer explicit checks here, but we're using this return value as a boolean and so C boolean semantics (0 is false, non-zero is true) are fine.  This test makes me wonder what the other return values are.

File: src/lkl/virtio_blkdev.c [hunk 2]
```diff
@@ -42,14 +42,17 @@ void lkl_add_disks(
         sgxlkl_enclave_state.shared_memory.virtio_blk_dev_mem
             [sgxlkl_enclave_state.disk_state[0].host_disk_index];
     int mmio_size = VIRTIO_MMIO_CONFIG + root_dev->config_len;
-    lkl_virtio_dev_setup(root_dev, mmio_size, lkl_deliver_irq);
+    if (lkl_virtio_dev_setup(root_dev, mmio_size, lkl_deliver_irq) != 0)
```
Comment (davidchisnall) [inline src/lkl/virtio_blkdev.c:45]: As above.

File: src/lkl/virtio_blkdev.c [hunk 3]
```diff
@@ -42,14 +42,17 @@ void lkl_add_disks(
...
+        return -1;
 
     for (size_t i = 0; i < num_mounts; ++i)
     {
         struct virtio_dev* dev =
             sgxlkl_enclave_state.shared_memory.virtio_blk_dev_mem
                 [sgxlkl_enclave_state.disk_state[i + 1].host_disk_index];
         int mmio_size = VIRTIO_MMIO_CONFIG + dev->config_len;
-        lkl_virtio_dev_setup(dev, mmio_size, lkl_deliver_irq);
+        if (lkl_virtio_dev_setup(dev, mmio_size, lkl_deliver_irq) != 0)
```
Comment (davidchisnall) [inline src/lkl/virtio_blkdev.c:54]: And here.

**Review by davidchisnall — APPROVED** [review]
Minor comments, happy for this to be merged once those are addressed.

**Review by paulcallen — APPROVED** [review]

File: src/lkl/setup.c (hunk 1 above)
Comment (Francis-Liu) [reply src/lkl/setup.c:864]: Sorry @davidchisnall I replied your comment using VScode GitHub plugin, but apparently the comment didn't go through to github.

I saw in sgxlkl return code 0 is used as SUCCESS. And -1, 1, or -SOME_ERROR_CODE are used on failure. Which gave me an impression that it follows Unix program's convention (return 0 on success). I think if we use non-zero as success, then we need to check against specific error code. What do you think?

This PR was merged, if you insist we use non-zero as success, then I can make the change in my next PR.

**Pull request merged**